sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -91,14 +91,25 @@ object SQLExecution {
         var ex: Option[Throwable] = None
         val startTime = System.nanoTime()
         try {
-          sc.listenerBus.post(SparkListenerSQLExecutionStart(
+
+          // queryExecution.executedPlan may throw an exception and made the send event message
+          // failed. we should send SparkListenerSQLExecutionStart event to listener bus when we
+          // catch an exception.
+          val sparkPlanInfo = try {
+            SparkPlanInfo.fromSparkPlan(queryExecution.executedPlan)
+          } catch {
+            case e: Exception =>
+              null
+          }
+
+          sparkSession.listenerManager.post(SparkListenerSQLExecutionStart(
```

Comment:
We send `SparkListenerSQLExecutionStart` event through `sparkSession.listenerManager.post`, this method will only send SQL executions event to the listener manager created by the SQL launched session, not all of the listener managers.